prov/gni/src/gnix_fabric.c

```diff
@@ -501,10 +501,14 @@ static int _gnix_ep_getinfo(enum fi_ep_type ep_type, uint32_t version,
 				gnix_info->domain_attr->data_progress =
 					hints->domain_attr->data_progress;
 
+
 			switch (hints->domain_attr->mr_mode) {
 			case FI_MR_UNSPEC:
 			case FI_MR_BASIC:
-				gnix_info->domain_attr->mr_mode = FI_MR_BASIC;
+				if (FI_VERSION_GE(version, FI_VERSION(1, 5))) {
+					hints->domain_attr->mr_mode =
+						OFI_MR_BASIC_MAP;
+				}
```

Comment:
The check in ofi_check_mr_mode, looks for the same bits set in the hints as in the provider mr_mode. The check is:
`return ((user_mode & prov_mode) == prov_mode) ? 0 : -FI_ENODATA;`